### PR TITLE
feat (ui): add banner to refresh UI

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: ui/src/assets/i18n/fr.json [hunk 1]
```diff
@@ -358,6 +358,8 @@
   "variable_updated" : "Variable mise à jour",
   "variable_value" : "Valeur",
 
+  "ui_updated" : "L'interface vient d'être mise à jour. Merci de cliquer dans cette zone pour rafraichir votre page.",
```
Comment (yesnault) [inline ui/src/assets/i18n/fr.json]: rafraîchir

File: ui/src/assets/i18n/en.json [hunk 2]
```diff
@@ -358,6 +358,8 @@
   "variable_updated" : "Variable updated",
   "variable_value" : "Value",
 
+  "ui_updated" : "UI has just been updated. Please click on this zone to refresh your page.",
```
Comment (fsamin) [inline ui/src/assets/i18n/en.json]: Please click here

File: ui/src/assets/i18n/fr.json (hunk 1 above)
Comment (fsamin) [inline ui/src/assets/i18n/fr.json]: Merci de cliquer ici
rafraîchir

File: ui/src/app/app.component.ts [hunk 3]
```diff
@@ -60,18 +59,28 @@ export class AppComponent  implements OnInit, OnDestroy {
                 this.isConnected = true;
                 this.startWarningWorker();
             }
+            this.startVersionWorker();
         });
     }
 
+    stopVersionWorker(): void {
```
Comment (bnjjj) [inline ui/src/app/app.component.ts]: Why not use a more generic function that take in parameter the name of the worker in order to have just one function and don't maintain 2 different functions that make the same things ?

File: ui/src/app/app.component.ts [hunk 4]
```diff
@@ -81,17 +90,44 @@ export class AppComponent  implements OnInit, OnDestroy {
...
             }
         });
     }
+
+    startVersionWorker(): void {
+        this.stopVersionWorker();
+        this.versionWorker = new CDSWorker('./assets/worker/web/version.js');
+        this.versionWorker.start({});
+        this.versionWorker.response().subscribe( msg => {
+            if (msg !== null) {
```
Comment (bnjjj) [inline ui/src/app/app.component.ts]: if msg == undefined it enters in the condition. Is it the comportement desired ?

**Comment by yesnault** [comment]
http://www.academie-francaise.fr/sites/academie-francaise.fr/files/rectifications_1990.pdf :-)

File: ui/src/assets/i18n/fr.json (hunk 1 above)
Comment (sguiheux) [inline ui/src/assets/i18n/fr.json]: french reform of 1990 says that you should forget ^ on i and u :)
http://www.academie-francaise.fr/sites/academie-francaise.fr/files/rectifications_1990.pdf

File: ui/src/app/app.component.ts [hunk 5]
```diff
@@ -106,11 +94,11 @@ export class AppComponent  implements OnInit {
     }
 
     startVersionWorker(): void {
-        this.stopVersionWorker();
+        this.stopWorker(this.versionWorker, this.versionWorkerSubscription);
         this.versionWorker = new CDSWorker('./assets/worker/web/version.js');
         this.versionWorker.start({});
         this.versionWorker.response().subscribe( msg => {
-            if (msg !== null) {
+            if (msg !== null && msg !== undefined) {
```
Comment (bnjjj) [inline ui/src/app/app.component.ts]: Not necessary to change but good to know but if you write `if (msg != null)` it catch undefined and null.

**Pull request merged**